web/src/settings_components.ts

```diff
@@ -966,6 +966,16 @@ export function populate_data_for_request(
                     });
                     continue;
                 }
+
+                if (property_name === "org_join_restrictions") {
+                    data = {
+                        ...data,
+                        ...settings_data.get_request_data_for_org_join_restrictions(
+                            input_value.toString(),
```

Comment:
There is still some bug here probably due to us sending two values to updates in the request particularly when updating from "Restricted to a list of domains" to "Don't allow disposable email addresses".

We discussed something similar that happens when updating two different settings in a single request and we did not find a clear for why it was happening. And this is a similar case where there is only one element in UI but it depends on two fields in database. Will look at this later again.